qa/rpc-tests/limits.py

```diff
@@ -0,0 +1,141 @@
+                outp[addrs[(count + x) % len(addrs)]] = payamt
+                #outscript = self.wastefulOutput(addrs[(count+x)%len(addrs)])
+                #outscripthex = hexlify(outscript).decode("ascii")
+                #outp[outscripthex] = payamt
+            if data:
+                outp["data"] = data
+            txn = createrawtransaction([utxo], outp, createWastefulOutput)
+            # txn2 = node.createrawtransaction([utxo], outp)
+            signedtxn = node.signrawtransaction(txn)
+            size += len(binascii.unhexlify(signedtxn["hex"]))
+            node.sendrawtransaction(signedtxn["hex"])
+        logging.info("%d tx %d length" % (count, size))
+        decimal.getcontext().prec = decContext
+        return (count, size)
+
+    def makeUtxos(self, node, addrs, NUM_BLOCKS):
+
+        outp = {}
+        for a in addrs:
+            outp[a] = 0.1
+
+        for x in range(0,NUM_BLOCKS):
+            tx = node.sendmany("", outp)
+            # print(tx)
+
+        node.generate(1)
+        self.sync_blocks()
+
+    def run_test (self):
+
+        TEST_BLOCK_SIZE=128000000
+        [ x.set("net.excessiveBlock=%d" % (TEST_BLOCK_SIZE+20000)) for x in self.nodes]
+        [ x.set("mining.blockSize=%d" % TEST_BLOCK_SIZE) for x in self.nodes]
+        [ x.set("mining.dataCarrierSize=64000") for x in self.nodes ] # allow lots of data
+        self.testBigSubmitBlock( TEST_BLOCK_SIZE, 100, 30)
+
+    def testBigSubmitBlock(self, TEST_BLOCK_SIZE, NUM_ADDRS, NUM_BLOCKS):
+        # generate enough blocks so that nodes[0] has a balance
+        self.sync_blocks()
+        self.nodes[0].generate(100+NUM_BLOCKS)
+        self.sync_blocks()
+        logging.info("generated initial blocks")
+        
+        self.nodes[0].keypoolrefill(NUM_ADDRS)
+        addrs = [self.nodes[0].getnewaddress() for _ in range(NUM_ADDRS)]
+        logging.info("generated addresses")
+
+        self.makeUtxos(self.nodes[0], addrs, NUM_BLOCKS)
+        logging.info("split utxos")
+
+        peers = self.nodes[0].getpeerinfo()
+        for p in peers:
+            self.nodes[0].disconnectnode(p["addr"])
+
+        legacyAddrs = [ self.nodes[0].getaddressforms(x)["legacy"] for x in addrs ]
+        self.generateTx(self.nodes[0], TEST_BLOCK_SIZE, legacyAddrs, "01"*64000)
+        logging.info("generated enough tx")
+
+        bigBlockHash = self.nodes[0].generate(1)[0]
+        logging.info("generated block %s" % str(bigBlockHash))
+        bigBlockHex = self.nodes[0].getblock(bigBlockHash, False)
+        logging.info("got block hex, length %d" % len(bigBlockHex))
+        n1blockcount = self.nodes[1].getblockcount()
+        ret = self.nodes[1].submitblock(bigBlockHex)
+        assert ret != "duplicate"
+        assert_equal(n1blockcount + 1, self.nodes[1].getblockcount())
+        assert_equal(self.nodes[0].getblockcount(), self.nodes[1].getblockcount())
+        return True
+
+
+if __name__ == '__main__':
+    MyTest ().main ()
```

Comment:
nit: sometime a space is placed between function name and brackets sometime is not

Reply:
good catch.  I will autopep8 the file